sympy/core/kind.py

```diff
@@ -187,6 +188,12 @@ def __repr__(self):
 BooleanKind = _BooleanKind()
 
 
+class _Dispatcher(Dispatcher):
+    def add(self, signature, func, on_ambiguity=ambiguity_warn):
+        super().add(signature, func)
+        super().reorder(on_ambiguity=on_ambiguity)
```

Comment:
Explicitly calling `reorder` here will probably affect the import time. The third party package reorders lazily by default and doesn't have an `on_ambiguity` parameter to `add`. For now I'm just trying to confirm that everything is working. Some code in SymPy especially for `KindDispatcher` seems to use `on_ambiguity` in nontrivial ways so it isn't clear yet that we could just remove that argument.